pkg/cli/pid.go

```diff
@@ -0,0 +1,58 @@
+package cli
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+)
+
+type ProcessSerialization interface {
+	Acquire() error
+	Release() error
+}
+
+type PIDFile struct {
+	file *os.File
+	fd   int
+}
+
+func NewPIDFile(path string) (*PIDFile, error) {
+	file, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE, 0644)
+	if err != nil {
+		return nil, err
+	}
+	return &PIDFile{file: file, fd: int(file.Fd())}, nil
+}
+
+func (p *PIDFile) Acquire() error {
+	funcs := []func() error{
+		func() error { return syscall.Flock(p.fd, syscall.LOCK_EX) },
+		p.writePID,
+		p.file.Sync,
+	}
+	return p.executeFuncs(funcs)
+}
+
+func (p *PIDFile) Release() error {
+	funcs := []func() error{
+		func() error { return syscall.Flock(p.fd, syscall.LOCK_UN) },
+		p.file.Close,
+		func() error { return os.Remove(p.file.Name()) },
+	}
+	return p.executeFuncs(funcs)
+}
+
+func (p *PIDFile) writePID() error {
+	pid := os.Getpid()
+	_, err := p.file.Write([]byte(fmt.Sprintf("%d", pid)))
```

Comment:
```suggestion
	_, err := p.file.WriteString(fmt.Sprintf("%d", pid))
```
(double check this compiles but I think it's available here)